SCClassLibrary/Common/Control/Server.sc

```diff
@@ -892,7 +948,11 @@ Server {
 	}
 
 	freeAll {
-		this.sendMsg("/g_freeAll", 0);
+		if (clientID == 0) {
+			this.sendMsg("/g_freeAll", 0);
+		} {
+			this.sendMsg("/g_freeAll", this.defaultGroupID);
```

Comment:
this seems a bit odd, so needs a comment, or separate methods that make explicit what is happening here.

Something like this would be clearer:
```supercollider
Server {
	
	var <>alwaysFreeRoot = false;

	shouldFreeAll { 
		^this.alwaysFreeRoot or: { clientID == 0 }
	}
	
	freeAll {
		if(this.shouldFreeAll) {
			this.freeSynthsInRootGroup
		} {
			this.freeSynthsInDefaultGroup
		};
		this.sendMsg("/clearSched");
 		this.initTree;
	}
	
	freeSynthsInRootGroup {
		this.sendMsg("/g_freeAll", 0)
	}
	
	freeSynthsInDefaultGroup {
		this.sendMsg("/g_freeAll", this.defaultGroupID)
	}
	
}
```